tests/test_streaming_download_manager.py

```diff
@@ -488,170 +479,183 @@ def test_xrelpath(input_path, start_path, expected_path):
-
-
-@pytest.mark.parametrize(
-    "input_path, expected_path",
-    [
-        (Path(__file__).resolve(), Path(__file__).resolve().parent),
-        (Path("https://host.com/archive.zip"), Path("https://host.com")),
-        (
-            Path("zip://file.txt::https://host.com/archive.zip"),
-            Path("zip://::https://host.com/archive.zip"),
-        ),
-        (
-            Path("zip://folder/file.txt::https://host.com/archive.zip"),
-            Path("zip://folder::https://host.com/archive.zip"),
-        ),
-    ],
-)
-def test_xpathparent(input_path, expected_path):
-    output_path = xPath(input_path).parent
-    output_path = _readd_double_slash_removed_by_path(output_path.as_posix())
-    assert output_path == _readd_double_slash_removed_by_path(expected_path.as_posix())
-
-
-@pytest.mark.parametrize(
-    "input_path, expected",
-    [
-        ("zip://file.txt::https://host.com/archive.zip", "file.txt"),
-        ("datasets/file.txt", "file.txt"),
-        ((Path().resolve() / "file.txt").as_posix(), "file.txt"),
-    ],
-)
-def test_xpathname(input_path, expected):
-    assert xPath(input_path).name == expected
-
-
-@pytest.mark.parametrize(
-    "input_path, expected",
-    [
-        ("zip://file.txt::https://host.com/archive.zip", "file"),
-        ("file.txt", "file"),
-        ((Path().resolve() / "file.txt").as_posix(), "file"),
-    ],
-)
-def test_xpathstem(input_path, expected):
-    assert xPath(input_path).stem == expected
-
-
-@pytest.mark.parametrize(
-    "input_path, expected",
-    [
-        ("zip://file.txt::https://host.com/archive.zip", ".txt"),
-        ("file.txt", ".txt"),
-        ((Path().resolve() / "file.txt").as_posix(), ".txt"),
-    ],
-)
-def test_xpathsuffix(input_path, expected):
-    assert xPath(input_path).suffix == expected
-
-
-@pytest.mark.parametrize(
-    "input_path, suffix, expected",
-    [
-        ("zip://file.txt::https://host.com/archive.zip", ".ann", "zip://file.ann::https://host.com/archive.zip"),
-        ("file.txt", ".ann", "file.ann"),
-        ((Path().resolve() / "file.txt").as_posix(), ".ann", (Path().resolve() / "file.ann").as_posix()),
-    ],
-)
-def test_xpath_with_suffix(input_path, suffix, expected):
-    assert xPath(input_path).with_suffix(suffix) == xPath(expected)
+class TestxPath:
+    @pytest.mark.parametrize(
+        "input_path, expected_path",
+        [
+            ("https://host.com/archive.zip", "https://host.com/archive.zip"),
+            ("zip://file.txt::https://host.com/archive.zip", "zip://file.txt::https://host.com/archive.zip"),
+            ("zip://dir/file.txt::https://host.com/archive.zip", "zip://dir/file.txt::https://host.com/archive.zip"),
+            ("file.txt", "file.txt"),
+            (str(Path().resolve() / "file.txt"), (Path().resolve() / "file.txt").as_posix()),
+        ],
+    )
+    def test_xpath_as_posix(self, input_path, expected_path):
+        assert xPath(input_path).as_posix() == expected_path
```

Comment:
Maybe you can add a similar test for str(xPath()) ?
```suggestion
        assert xPath(input_path).as_posix() == expected_path

    @pytest.mark.parametrize(
        "input_path",
        [
            "https://host.com/archive.zip",
            "zip://file.txt::https://host.com/archive.zip",
            "zip://dir/file.txt::https://host.com/archive.zip",
            "file.txt",
            str(Path().resolve() / "file.txt"),
        ],
    )
    def test_xpath_str(self, input_path):
        assert str(xPath(input_path)) == input_path
```

Reply:
Well seen! ;)